### ISSUE TITLE
GraphQL 0.8.1 support?

### ISSUE THREAD

**Issue by mnpenner** [jakepusateri/graphql-list-fields#1]
Does this work with 0.8.1? I'm getting this error message:

> Cannot read property 'selectionSet' of undefined

When I use it like this:

```js
const QueryType = new GraphQLObjectType({
    name: 'Query',
    description: 'The root of all... queries',
    fields: () => ({
        ...
        
        calendars: {
            type: require('./CalendarApi'),
            resolve: (parent, args, context, info) => {
                let fields = getFieldNames(info);
                debug(fields);
            }
        }
    }),
});
```

**Comment by mnpenner** [comment]
It appears that `fieldASTs` has been renamed to `fieldNodes`. Here's my implementation that works with 0.8.1, but is missing support for the skip/exclude directives:

```js
const Kinds = require('graphql/language/kinds');

/**
 *
 * @param {Object} context
 * @param {Array} asts
 * @returns {Set}
 */
module.exports = function getFieldNames(context, asts=context.fieldNodes) {
    if (!Array.isArray(asts)) {
        asts = [asts];
    }

    return asts.map(ast => ast.selectionSet.selections).reduce(Array.prototype.concat).reduce((fields, selection) => {
        switch(selection.kind) {
            case Kinds.FIELD:
                return fields.add(selection.name.value);
            case Kinds.INLINE_FRAGMENT:
                return fields.unionWith(getFieldNames(context, selection));
            case Kinds.FRAGMENT_SPREAD:
                return fields.unionWith(getFieldNames(context, context.fragments[selection.name.value]));
            default:
                throw new Error(`Unsupported selection kind: ${selection.kind}`)
        }

    }, new Set());
};
```

**Comment by mccluresc** [comment]
The issue here is that fieldASTs is renamed to fieldNodes

**Comment by mnpenner** [comment]
@mccluresc Yeah... I think that was the only change.

**Comment by jakepusateri** [comment]
Published a fix in 1.1.0
